### PR TITLE
Add @trimblemaps/trimblemaps-js Types

### PR DESCRIPTION
**Pull request by PaitoAnderson**
A lot of this was copied directly from mapbox-gl to support @trimblemaps/trimblemaps-js. I'm not sure this is the best solution as it may be difficult to maintain, open to suggestions.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
Is this a plugin or extension to `mapbox-lg`? (so it can be used in place of original `mapbox-gl` package.

**Comment by PaitoAnderson** [comment]
It's more of a fork of mapbox-gl with additional features, and only built for one mapping service Trimble Maps.

**Comment by peterblazejewicz** [comment]
ok, can it be written as augmented features on `mapbox-lg`, say here is how this is done for a leaflet:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/leaflet-freehandshapes/index.d.ts
this module in PR is more like https://github.com/mapbox/mapbox-gl-geocoder
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mapbox__mapbox-gl-geocoder/index.d.ts
in the aspects of: 
- it exports it's own namespace `TrimbleMaps`
- it uses UMD package.
sorry for the questions, I cannot see original code, so we need to rely on your input, thx!

**Comment by PaitoAnderson** [comment]
Hi @peterblazejewicz,

Unfortunately, it's not open source, and I can't see the code either which is a disadvantage. However based on the docs they say it's based on Mapbox GLJS (v1.12) which is after mapbox-gl/v1 and before the current version (2.4), which complicates things a bit when it comes to augmenting features as we'd need to omit<> some things when extending interfaces which could get ugly.

Would you be OK accepting this as-is to be maintained separately?

**Comment by peterblazejewicz** [comment]
> Would you be OK accepting this as-is to be maintained separately?

Yes, absolutely.

File: types/trimblemaps__trimblemaps-js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,2817 @@
+// Type definitions for @trimblemaps/trimblemaps-js 3.0
+// Project: https://developer.trimblemaps.com/maps-sdk/api/
+// Definitions by: Paito Anderson <https://github.com/PaitoAnderson>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// THIS FILE WAS PARTIALLY COPIED FROM mapbox-gl
+
+/// <reference types="geojson" />
+
```
Comment (peterblazejewicz) [inline types/trimblemaps__trimblemaps-js/index.d.ts:9]:
```suggestion

export as namespace TrimbleMaps;

```
I think you need this one, global script support is explicitly mentioned on api docs:
https://developer.trimblemaps.com/maps-sdk/guide/getting-started/
and as I said, it's visible in UMD-type of the package itself:
https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#identifying-a-umd-library

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
As discussed, I think we agree this could be improved, observing how leaflet or mapbox are extended. But for now it serves the purpose for TS users.

@PaitoAnderson thx!

**Comment by PaitoAnderson** [comment]
Thanks @peterblazejewicz - Appreciate you taking the time to review and share how this can be improved. If I have time I'll take a look at implemeting the extension way in a future PR.

Ready to merge

**Comment by PaitoAnderson** [comment]
Ready to merge

**Pull request merged**